decidim-core/app/services/decidim/onboarding_manager.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This class is used to manage the onboarding funnel for users that
+  # intended to perform an action in the platform but it requires some
+  # verification steps to be completed.
+  #
+  # The flow is as follows:
+  #
+  # 1. A visitor (logged out) tries to perform an action that requires to be logged in
+  # 2. The intended action is stored in a cookie (onboarding key)
+  # 3. The visitor logs in and the onboarding cookie data is merged into the User#extended_data column.
+  # 4. After logging in, the user is redirected to the onboarding pending authorizations page
+  #    if there is any pending action stored in the User#extended_data column
+  # 5. The user completes the required authorizations
+  # 6. When the onboarding authorizations page detects that all the verification steps have been
+  #    completed, it will remove the pending action from the User#extended_data column and redirect
+  #    the user to the intended action.
+  #
+  class OnboardingManager
+    DATA_KEY = "onboarding"
+
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    # Checks if the onboarding data has an action and a model.
+    #
+    # Returns a boolean
+    def valid?
+      return if action.blank?
+
+      permissions_holder.present?
+    end
+
+    # Returns the action to be performed.
+    #
+    # Returns a string
+    def action
+      onboarding_action
+    end
+
+    # Returns the translation of the action to be performed if found, action
+    # elsewhere
```

Comment:
I think this comment is incomplete? (", action elsewhere" seems wrong)

Reply:
comment improved